src/libsystemd-network/radv-internal.h

```diff
@@ -98,6 +98,7 @@ struct sd_radv {
 
         int ifindex;
         char *ifname;
+        struct in6_addr ipv6ll;
```

Comment:
@poettering 
Yeah, potentially, the interface may have multiple IPv6LL addresses.
But, that's quite rare, and at least the kernel assigns only one IPv6LL address, so the concerned situation occurs only when a user explicitly and manually configure an IPv6LL address with `Address=` (or of course by `ip addr` command).
This is a kind of safe guard for user misconfiguration. I think that's overkill to support such situation at least now.
Note, systemd-networkd currently does not support multiple IPv6LL addresses gracefully, and supporting that requires huge rework. I'd like to merge this as is at least now, and rework later if necessary.

Reply:
> I think that's overkill to support such situation at least now

Agreed. Then again what I was trying to say is that if networkd didn't implement the "MAY" part from  https://datatracker.ietf.org/doc/html/rfc4861#section-6.2.6
> A router MAY choose to unicast the
   response directly to the soliciting host's address (if the
   solicitation's source address is not the unspecified address), but
   the usual case is to multicast the response to the all-nodes group.

it wouldn't have to worry about unicast advertisements at all. It would have to reset the timer though
> In the latter case, the interface's interval timer is reset to a new
   random value, as if an unsolicited advertisement had just been sent
   (see [Section 6.2.4](https://datatracker.ietf.org/doc/html/rfc4861#section-6.2.4)).

and that part isn't implemented as far as I can tell.

Reply:
The latter case will be addressed in another later PR. The fix is already queued in the network-next branch.

Reply:
Got it. I took a look at the next branch yesterday and it wasn't there. I should have taken a look today probably :-)

Reply:
Ah, may not be there yet. I may forgot to push.